verify/verify_test.go

```diff
@@ -18,6 +24,31 @@ type VerifySuite struct{}
 
 var _ = Suite(&VerifySuite{})
 
+type ecdsaSigner struct {
+	*ecdsa.PrivateKey
+}
+
+func (s ecdsaSigner) PublicData() *data.Key {
+	pub := s.Public().(*ecdsa.PublicKey)
+	return &data.Key{
+		Type:  data.KeyTypeECDSA_SHA2_P256,
+		Value: data.KeyValue{Public: elliptic.Marshal(pub.Curve, pub.X, pub.Y)},
+	}
+}
+
+func (s ecdsaSigner) Sign(rand io.Reader, msg []byte, opts crypto.SignerOpts) ([]byte, error) {
+	hash := sha256.Sum256(msg)
+	return s.PrivateKey.Sign(rand, hash[:], crypto.SHA256)
+}
+
+func (s ecdsaSigner) ID() string {
+	return s.PublicData().ID()
+}
+
+func (ecdsaSigner) Type() string {
+	return data.KeyTypeECDSA_SHA2_P256
+}
+
```

Comment:
Yeah. This patch intentionally implements only verification in the exposed code. This is in preparation for supporting ECDSA signing keys via PKCS11 in external devices like the Yubikey and HSMs. If you can store keys locally, Ed25519 should be used. We're only adding support for ECDSA because the dedicated hardware devices do not support Ed25519.